proxy/proxy.go

```diff
@@ -25,6 +29,28 @@ var (
 	listenAddress = kingpin.Flag("web.listen-address", "Address to listen on for proxy and client requests.").Default(":8080").String()
 )
 
+var (
+	httpAPICounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "pushproxy_http_requests_total",
+			Help: "Number of http api requests.",
+		},
+		[]string{"code", "method", "path"},
```

Comment:
The method doesn't tell us anything here, they're always the same.

You should also initilise the children which you know will exist.